google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -86,6 +98,14 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
+    // The environment will be null for newly deserialized deployment sources to ensure freshness.
+    // In this case, we need to reload the environment.
+    if (environment == null) {
+      environment = AppEngineProjectService.getInstance().getModuleAppEngineEnvironment(
```

Comment:
So, IIUC, we will always be recalculating the environment for maven build sources, which might involve an expensive disk query. I'm wondering if there is a way to save some of this work, e.g., by caching the environment in the persisted settings as we appeared to be doing before. I wouldn't think this is critical, though.